### PR TITLE
[Improvement] [High Priority] Swap UI - Release 2

### PR DESCRIPTION
**Pull request by nguyenhuudungz**

### CODE REVIEW

File: src/components/SwapForm/SwapModal/SwapBrief.tsx [hunk 1]
```diff
@@ -26,76 +32,134 @@ const TruncatedText = styled(Text)`
...
 
   const renderOutputAmount = () => {
     if (isLoading) {
       return (
         <Skeleton
-          width="120px"
+          width="108px"
           // there's border of 1px
-          height="22px"
+          height="26.5px"
```
Comment (nguyenhoaidanh) [inline src/components/SwapForm/SwapModal/SwapBrief.tsx:77]: ugly number

File: src/components/SwapForm/SwapModal/SwapModalAreYouSure.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+`
+
+const StyledCloseIcon = styled(X)`
+  height: 28px;
+  width: 28px;
+  :hover {
+    cursor: pointer;
+  }
+
+  > * {
```
Comment (nguyenhoaidanh) [inline src/components/SwapForm/SwapModal/SwapModalAreYouSure.tsx:40]: `color: theme.text` is not working?

File: src/components/SwapForm/hooks/useGetRoute.ts [hunk 3]
```diff
@@ -81,10 +81,12 @@ const useGetRoute = (args: ArgsGetRoute) => {
     })
 
     const url = `${aggregatorDomain}/${NETWORKS_INFO[chainId].aggregatorRoute}/api/v1/routes`
+    // const url = `http://localhost:8014/${NETWORKS_INFO[chainId].aggregatorRoute}/api/v1/routes`
```
Comment (nguyenhoaidanh) [inline src/components/SwapForm/hooks/useGetRoute.ts:84]: rm

File: src/pages/SwapV2/index.tsx [hunk 4]
```diff
@@ -676,6 +641,8 @@ export default function Swap() {
     })
   }
 
+  const [isShowDegenBanner, setShowDegenBanner] = useState(true)
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV2/index.tsx:644]: default show ?

File: src/pages/SwapV3/HeaderRightMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    setShowHeaderMenu(false)
+  }
+  const onClickMoreButton = () => {
+    setShowHeaderMenu(prev => !prev)
+  }
+
+  return (
+    <SwapFormActions onMouseEnter={onMouseEnterMenu} onMouseLeave={onMouseLeaveMenu}>
+      {isShowHeaderMenu && (
+        // {true && (
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV3/HeaderRightMenu.tsx:124]: rm

File: src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx [hunk 6]
```diff
@@ -59,39 +67,38 @@ type Props = {
...
   const priceImpactFromBuild = buildResult?.data
     ? calculatePriceImpact(Number(buildResult?.data?.amountInUsd || 0), Number(buildResult?.data?.amountOutUsd || 0))
     : undefined
 
   const priceImpactResult = checkPriceImpact(priceImpactFromBuild)
 
-  const outputAmountChange = Number(buildResult?.data?.outputChange?.amount) || 0
+  const outputChangePercent = Number(buildResult?.data?.outputChange?.percent) || 0
+  const formattedOutputChangePercent =
+    -0.001 < outputChangePercent && outputChangePercent < 0 ? '< -0.001' : outputChangePercent.toFixed(3)
```
Comment (nguyenhoaidanh) [inline src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx:96]: wrong condition? 
outputChangePercent < -0.001

File: src/components/SwapForm/hooks/useBuildRoute.tsx [hunk 7]
```diff
@@ -44,7 +45,10 @@ const useBuildRoute = (args: Args) => {
     }
 
     const payload: BuildRoutePayload = {
-      routeSummary,
+      routeSummary: {
+        ...routeSummary,
+        // amountOut: (Number(routeSummary.amountOut) * 2).toString(),
```
Comment (nguyenhoaidanh) [inline src/components/SwapForm/hooks/useBuildRoute.tsx:50]: rm

File: src/components/TransactionConfirmationModalV3/index.tsx [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+            <Trans>Confirm this transaction in your wallet</Trans>
+          </Text>
+          <HurryUpBanner startedTime={startedTime} />
+        </AutoColumn>
+      </Section>
+    </Wrapper>
+  )
+}
+
+function AddTokenToInjectedWallet({ token, chainId }: { token: Token; chainId: ChainId }) {
```
Comment (nguyenhoaidanh) [inline src/components/TransactionConfirmationModalV3/index.tsx:93]: duplicate component

File: src/components/SwapForm/SwapModal/SwapBrief.tsx (hunk 1 above)
Comment (nguyenhuudungz) [reply src/components/SwapForm/SwapModal/SwapBrief.tsx:77]: But it's worked.

Comment (nguyenhuudungz) [reply src/components/SwapForm/SwapModal/SwapBrief.tsx:77]: 0.5px is exactly here.

File: src/components/SwapForm/SwapModal/SwapModalAreYouSure.tsx (hunk 2 above)
Comment (nguyenhuudungz) [reply src/components/SwapForm/SwapModal/SwapModalAreYouSure.tsx:40]: Tbh, I don't know why I put it here. But it's work.

File: src/components/SwapForm/hooks/useGetRoute.ts (hunk 3 above)
Comment (nguyenhuudungz) [reply src/components/SwapForm/hooks/useGetRoute.ts]: Done.

File: src/pages/SwapV2/index.tsx (hunk 4 above)
Comment (nguyenhuudungz) [reply src/pages/SwapV2/index.tsx:644]: Yes.

File: src/pages/SwapV3/HeaderRightMenu.tsx (hunk 5 above)
Comment (nguyenhuudungz) [reply src/pages/SwapV3/HeaderRightMenu.tsx:124]: Done.

File: src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx (hunk 6 above)
Comment (nguyenhuudungz) [reply src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx]: Tks, fixed.

File: src/components/SwapForm/hooks/useBuildRoute.tsx (hunk 7 above)
Comment (nguyenhuudungz) [reply src/components/SwapForm/hooks/useBuildRoute.tsx:50]: Done.

File: src/components/SwapForm/SwapModal/SwapModalAreYouSure.tsx (hunk 2 above)
Comment (nguyenhoaidanh) [reply src/components/SwapForm/SwapModal/SwapModalAreYouSure.tsx:40]: pls update color: theme.text

File: src/components/SwapForm/SwapModal/SwapBrief.tsx (hunk 1 above)
Comment (nguyenhoaidanh) [reply src/components/SwapForm/SwapModal/SwapBrief.tsx:77]: how about 26px?
26px vs 26.5px, I think there is no difference

File: src/components/TransactionConfirmationModalV3/index.tsx (hunk 8 above)
Comment (nguyenhuudungz) [reply src/components/TransactionConfirmationModalV3/index.tsx]: Fixed.

File: src/components/SwapForm/hooks/useBuildRoute.tsx [hunk 9]
```diff
@@ -44,7 +45,9 @@ const useBuildRoute = (args: Args) => {
     }
 
     const payload: BuildRoutePayload = {
-      routeSummary,
+      routeSummary: {
+        ...routeSummary,
```
Comment (nguyenhoaidanh) [inline src/components/SwapForm/hooks/useBuildRoute.tsx:49]: why do we need copy here

**Comment by nguyenhoaidanh** [comment]
lgmt

**Review by nguyenhoaidanh — APPROVED** [review]

**Review by nguyenhoaidanh — DISMISSED** [review]

File: src/hooks/index.ts [hunk 10]
```diff
@@ -117,7 +119,7 @@ export function useWeb3React(key?: string): Web3ReactContextInterface<Web3Provid
     connector,
     library: library || provider,
     chainId: chainId || ChainId.MAINNET,
-    account,
+    account: searchParams.get('account') || account,
```
[the remark this answers is not in the document]
Comment (nguyenhuudungz) [reply src/hooks/index.ts:122]: When the query empty, then fallback to web3 account.

**Review by nguyenhoaidanh — APPROVED** [review]

File: src/components/SlippageControl/CustomSlippageInput.tsx [hunk 11]
```diff
@@ -148,7 +153,9 @@ const CustomSlippageInput: React.FC<Props> = ({ rawSlippage, setRawSlippage, isW
       return
     }
 
-    if (parsedValue > MAX_SLIPPAGE_IN_BIPS) {
+    const maxSlippage = isDegenMode ? MAX_DEGEN_SLIPPAGE_IN_BIPS : MAX_NORMAL_SLIPPAGE_IN_BIPS
+    if (parsedValue > maxSlippage) {
+      setTooltip(t`Max is ${formatSlippage(maxSlippage)}`)
```
Comment (viet-nv) [inline src/components/SlippageControl/CustomSlippageInput.tsx:158]: we dont need a state to manage tooltip message here. Just a variable that we can infer from `rawText` 
```
const maxSlippage = isDegenMode ? MAX_DEGEN_SLIPPAGE_IN_BIPS : MAX_NORMAL_SLIPPAGE_IN_BIPS
const tooltip = parseSlippageInput(rawText) > maxSlippage ? t`Max is ${formatSlippage(maxSlippage)}` : "

**Pull request merged**